modules/EnsEMBL/Web/Component/Gene/ComparaOrthologs.pm

```diff
@@ -74,53 +74,39 @@ sub content {
   my %orthologue_list;
   my %skipped;
 
-  my %not_seen = $self->_get_all_analysed_species($cdb);
+  my %species_to_show = $self->_get_all_analysed_species($cdb);
+  my $this_group      = $species_defs->STRAIN_GROUP;
 
-  # The %not_seen hash can either have the production name or species url as the code is shared across ensembl and all divisions of e!g.
-  # These use either production name or species url, so difficult to find out what is used by the %not_seen hash.
-  # In some case cases, such as pan compara, there can be a mix of production name and species url.
-  # Therefore, need to have two delete statements to make sure the species (with orthologues) get deleted properly.
-  delete $not_seen{$hub->species}; #deleting current species
-  delete $not_seen{lc($hub->species)};
-
-  for (keys %not_seen) {
-    #do not show non-strain species on strain view
-    if ($self->is_strain && !$species_defs->get_config($_, 'RELATED_TAXON')) {
-      delete $not_seen{$_};
-      delete $not_seen{lc $_};
-    }
-
-    #do not show strain species on main species view
-    if (!$self->is_strain && $species_defs->get_config($_, 'IS_STRAIN_OF')) {
-      delete $not_seen{$_};
-      delete $not_seen{lc $_};
-    }
-  }
+  ## Skip current species
+  delete $species_to_show{$species_defs->SPECIES_PRODUCTION_NAME};
 
   foreach my $homology_type (@orthologues) {
     foreach my $species (keys %$homology_type) {
-
-      #do not show strain species on main species view
-      if ((!$self->is_strain && $species_defs->get_config($species, 'IS_STRAIN_OF')) || ($self->is_strain && !$species_defs->get_config($species, 'RELATED_TAXON'))) {
-        delete $not_seen{$species};
-        delete $not_seen{lc $species};
-        next;
+      
+      my $prod_name     = $species_defs->get_config($species, 'SPECIES_PRODUCTION_NAME');
```

Comment:
Are we trying to avoid these calls by using production names? If so should we change https://github.com/Ensembl/ensembl-webcode/blob/release/104/modules/EnsEMBL/Web/Object/Gene.pm#L822 to use production name? Or is it going to be affected in many places?

Reply:
We probably should, but I think it can be left until 106 - I'll make a note.